internal/services/redisenterprise/validate/redis_enterprise_cluster_location.go

```diff
@@ -52,12 +52,14 @@ func friendlyValidRedisEnterpriseClusterLocations() []string {
 		"Central US EUAP",
 		"East Asia",
 		"East US",
+		"Germany West Central",
```

Comment:
@sinbai rather than continuing to add to this list can we remove this specific validation and use a regular `location` field instead?

Reply:
The code has been updated, could you please take another look? The test result are as follows.
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/39109137/d58e8800-1f3e-44f3-ae1b-7c8aa294644c)